cmd/root.go

```diff
@@ -45,8 +51,66 @@ var (
 	// SignalReceived is set to true when a signal causes us to exit. This makes
 	// determining the error message to show on exit much more easy.
 	SignalReceived = false
+
+	// profileCleanup is called after the root command is executed to
+	// cleanup a running cpu profile.
+	profileCleanup func()
 )
 
+// rootPreRun is executed before the root command runs and sets up cpu
+// profiling.
+func rootPreRun(*cobra.Command, []string) error {
```

Comment:
Would you mind linking in that this comes from: https://golang.org/pkg/runtime/pprof/#hdr-Profiling_a_Go_program